protocol/communities/manager.go

```diff
@@ -1845,7 +1845,38 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, request
 			return nil, err
 		}
 		requestToJoin.State = RequestToJoinStateAccepted
-		return requestToJoin, nil
+	}
+
+	// Save revealed addresses + signatures so they can later be added
+	// to the community member list when the request is accepted
+	if len(request.RevealedAccounts) > 0 {
+		err = m.persistence.SaveRequestToJoinRevealedAddresses(requestToJoin)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This looks a big backwards to me. Shouldn't we only save the revealed addresses, **after** we've verified the signatures?

Reply:
fixed